code/datums/components/dusting.dm

```diff
@@ -0,0 +1,249 @@
+		))
+	return ..()
+
+
+/datum/component/dusting/proc/attack_atom(datum/source, atom/attacked_atom, mob/living/user, params)
+	SIGNAL_HANDLER
+	if(consume_on_attack)
+		if(consume_atom(attacked_atom))
+			return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/datum/component/dusting/proc/on_attack_hand(datum/source, mob/user, list/modifiers)
+	SIGNAL_HANDLER
+	if(consume_on_attack_hand && isturf(atom_parent.loc)) //Prevent eating if it's in your inventory
+		consume_atom(user)
+
+/datum/component/dusting/proc/on_attackby(datum/source, obj/item/attacking_item, mob/user, params)
+	SIGNAL_HANDLER
+	if(consume_on_attackby && isturf(atom_parent.loc)) //Prevent eating if it's in your inventory
+		if(consume_atom(attacking_item))
+			return COMPONENT_NO_AFTERATTACK
+
+/datum/component/dusting/proc/on_pickup(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(consume_on_pickup)
+		if(consume_atom(user))
+			return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/datum/component/dusting/proc/on_bumped(datum/source, atom/bumped_atom)
+	SIGNAL_HANDLER
+	if(consume_on_bumped)
+		consume_atom(bumped_atom)
+
+/datum/component/dusting/proc/on_throw_impact(datum/source, hit_atom, throwingdatum)
+	SIGNAL_HANDLER
+	if(consume_on_throw)
+		consume_atom(hit_atom)
+
+/datum/component/dusting/proc/consume_atom(atom/consumed_atom, currently_consuming_contents)
+	if(consumed_atom.flags_1 & ignore_flags)
+		return
+	if(is_type_in_typecache(consumed_atom, ignore_subtypesof))
+		return
+	if((!consume_turfs && isturf(consumed_atom)) || currently_consuming_contents) //Don't consume turfs multiple times, that's bad juju
+		return
+	if(ismob(consumed_atom) && !isliving(consumed_atom)) //Don't kill ghosts/eyes somehow
+		return
+	if(ignoring_admin_messages == 1)
+		ignoring_admin_messages = 2
+		message_admins("[atom_parent] ([src.type]) has consumed multiple objects [ADMIN_JMP(atom_parent)].")
+
+	//First, consume the contents
+	if(consume_all_contents && !currently_consuming_contents)
+		ignoring_admin_messages = 1
+		for(var/atom/thing as anything in consumed_atom.get_all_contents())
+			if(thing != atom_parent)
+				consume_atom(thing, TRUE)
+		ignoring_admin_messages = 0
+	
+	//Next, consume the actual atom
+	if(ismob(consumed_atom))
+		var/mob/living/consumed_living = consumed_atom
+		if(consumed_living.status_flags & ignore_status_flags)
+			return
+		message_admins("[atom_parent] ([src.type]) has consumed [key_name_admin(consumed_living)] [ADMIN_JMP(atom_parent)].")
+		atom_parent.investigate_log("has consumed [key_name(consumed_living)].", INVESTIGATE_SUPERMATTER)
+		consumed_living.visible_message(span_danger("[consumed_living] slams into [atom_parent] inducing a resonance... [consumed_living.p_their()] body starts to glow and burst into flames before flashing into dust!"),
+			span_userdanger("You slam into [atom_parent] as your ears are filled with unearthly ringing. Your last thought is \"Oh, fuck.\""),
+			span_hear("You hear an unearthly noise as a wave of heat washes over you."))
```

Comment:
This isn't very modular

Reply:
It will soontm become modular because you mentioned that